Add support for Grafana annotations API (#322)
fluxcd/notification-controller: docs/spec/v1beta1/provider.md

```

### Grafana

> Sends notifications to [grafana annotations api](https://grafana.com/docs/grafana/latest/http_api/annotations/) 
> if grafana has authentication configured it is recommended to use secretRef and include the API key. 
> To enable the annotations on a Dashboard its needed to enable the annotations with Tag `flux` on the Dashboard settings by.
> This can be be performed by doing the following steps:
> * Annotations > Query > Enable Match any
> * Annotations > Query > Tags (Add Tag: `flux`)

Reviewer: ```suggestion
To send notifications to [Grafana annotations API](https://grafana.com/docs/grafana/latest/http_api/annotations/),
you have to enable the annotations on a Dashboard like so:

- Annotations > Query > Enable Match any
- Annotations > Query > Tags (Add Tag: `flux`)
```
